### DISCUSSION TITLE
Auth: Sending custom data via Third Party Providers

### DISCUSSION THREAD

**Discussion by Johann01** [supabase/supabase#3536] [General]
Hey Supabase Team,
maybe a newbie question, but I am trying to build something similar to the https://github.com/supabase/supabase/tree/master/examples/nextjs-slack-clone

I also have an app with RBAC.

And I want to create users, with the app_role 'buyer' or 'seller' based on the metadata you can provide from the client this already works with the auth.signUp(email, password), but is there a way to do that with OAuth and a third-party provider? 

```sql
-- inserts a row into public.users and assigns roles
create function public.handle_new_user()
returns trigger as $$
declare is_admin boolean;
begin
  insert into public.users (id, username)
  values (new.id, new.email);

  select count(*) = 1 from auth.users into is_admin;

  if position('buyer' in new.raw_user_meta_data->>'app_role') > 0 then
    insert into public.user_roles (user_id, role) values (new.id, 'buyer');
  elsif position('seller' in new.raw_user_meta_data->>'app_role') > 0 then
    insert into public.user_roles (user_id, role) values (new.id, 'seller');
  else
    insert into public.user_roles (user_id, role) values (new.id, 'buyer');
  end if;

  return new;
end;
$$ language plpgsql security definer;

-- trigger the function every time a user is created
create trigger on_auth_user_created
  after insert on auth.users
  for each row execute procedure public.handle_new_user();

```

**Comment by KevinGuerreroJimenez** [comment]
Same here, did you manage to solve it?

**Reply by Johann01** [reply]
Nope, I solved it by adding the logic in the client and updating the role of the user, once he performs a certain action in the client.

**Reply by mliakos** [reply]
Yeah, me too. This is really dirty though 😛 We should be able to intercept & provide some metadata before the request gets sent to supabase.

**Reply by MrVibe** [reply]
Same thing, I passed this to the client.
